Source/MQTTnet.Server/Status/MqttSessionStatus.cs

```diff
@@ -40,22 +41,64 @@ public Task DeleteAsync()
         return _session.DeleteAsync();
     }
 
-    public Task DeliverApplicationMessageAsync(MqttApplicationMessage applicationMessage)
+    /// <summary>
+    /// Delivers an application message immediately to the session.
+    /// </summary>
+    /// <param name="applicationMessage">The application message to deliver.</param>
+    /// <returns>
+    /// A task that represents the asynchronous operation.
+    /// The result contains the <see cref="InjectMqttApplicationMessageResult"/> that includes the packet identifier of the enqueued message.
+    /// </returns>
+    public async Task<InjectMqttApplicationMessageResult> DeliverApplicationMessageAsync(MqttApplicationMessage applicationMessage)
     {
         ArgumentNullException.ThrowIfNull(applicationMessage);
 
         var packetBusItem = new MqttPacketBusItem(MqttPublishPacketFactory.Create(applicationMessage));
         _session.EnqueueDataPacket(packetBusItem);
 
-        return packetBusItem.WaitAsync();
+        var mqttPacket = await packetBusItem.WaitAsync();
+
+        var result = new InjectMqttApplicationMessageResult()
+        {
+            PacketIdentifier = ((MqttPacketWithIdentifier)mqttPacket).PacketIdentifier
```

Comment:
When you create the PublishPacket above and keep the variable, there is no need to cast it here.

Reply:
Fair. Did it